internal/ethapi/api.go

```diff
@@ -1156,6 +1157,13 @@ func AccessList(ctx context.Context, b Backend, blockNrOrHash rpc.BlockNumberOrH
 		return nil, 0, nil, err
 	}
 
+	// Apply state overrides if provided
+	if stateOverrides != nil {
```

Comment:
Can you please move it to immediately after StateAndHeaderBy*? E.g. it might be possible that user overrides the nonce for which we take the default from db. This would create a nonce mismatch.

Reply:
fixed [62a0726](https://github.com/ethereum/go-ethereum/pull/31497/commits/62a07266b3f1ea307906669c77f6e386244bd9dd)